authentication.go

```diff
@@ -51,7 +51,7 @@ func (s *AuthenticationService) SetDigestAuth(username, password string) {
 // returns 401 Unauthorized and authType was set to authTypeDigest. The
 // resulting string is used to set the Authorization header before retrying
 // the request.
-func (s *AuthenticationService) digestAuthHeader(response *http.Response) (string, error) {
+func (s *AuthenticationService) digestAuthHeader(method string, response *http.Response) (string, error) {
```

Comment:
The `response` object contains the original request.
So you don`t have to pass the `method` here.
I think `response.Request.Method` should work here.
See https://golang.org/pkg/net/http/#Response

What do you think?